kernprof.py

```diff
@@ -151,10 +151,15 @@ def find_script(script_name):
 
 
 def main(args=None):
+    def strictly_positive(option, opt, value, parser):
+        if value <= 0:
+            raise optparse.OptionValueError("option %s: floating-point value must be > 0, got %s" % (opt, value))
+        setattr(parser.values, option.dest, value)
+
     if args is None:
         args = sys.argv
     usage = "%prog [-s setupfile] [-o output_file_path] scriptfile [arg] ..."
-    parser = optparse.OptionParser(usage=usage, version="%prog 1.0b2")
+    parser = optparse.OptionParser(usage=usage, version="%prog 1.1")
```

Comment:
Can this be changed to use the `__version__` attribute?

Reply:
There is no `__version__` for this script but I can add one.

Reply:
Can kernprof not import line_profiler and use its version and fallback to a special value if its not known (because of an ImportError)?

Reply:
I made a commit attempting to resolve the issue. I'm ready to merge after this is fixed.

Reply:
I don't understand the `try except`, how can it fail? The package should be installed anyway so if you're ok with that I would remove it, and unify the versions: just use line_profiler `__version__` for kernprof.

Reply:
Well, the kernprof modules doesn't depend on line_profiler, so I'm wary of adding in that dependency as a hard-dependency. I think its highly unlikely to fail, but I can see a few cases where it might: 

* the sys.path got messed with 
* kernprof is being invoked from outside the repo directory and line_profiler is not installed. 
* An older version of line_profiler is installed. 

These are rare cases, but I think its safe enough to have the try except. I think I'd rather keep it in. Could you revert the last commit? (Also can you correctly use `version=kernprof_version` as I seem to have forgotten to).

Reply:
I can revert it but I honestly think this introduces complexity for useless situations, as kernprof *is* a part of line_profiler:
- if you mess sys.path in such a way to make modules fail, it is expected for modules to fail
- line_profiler (and as such kernprof) should not be expected to work if it's not installed
- If an older version of line_profiler is installed, then an older version of kernprof is installed? I don't understand the problem.

If we keep the try except anyway, should I set the version to 'UNKNOWN' in the except?

Reply:
I do understand the useless complexity argument, and I'm not discounting it. I'm also considering the fact that I can't imagine all possible ways this could be used (personally, I don't even see why kernprof.py is necessary at all), so I err heavily on the side of taking steps to prevent any possible interruption to end-users of the library. While kernprof is part of the line_profiler package, it doesn't actually depend on it. Maintaining decoupled components within a library does have merit. 

That being said, you're right this wont be an issue in the vast majority of cases. If anyone does have an error they could either suffer in silence or raise an issue in this repo. 

At this current time I'd be fine with either: 

option (a): 
```python
try:	    
    from line_profiler import __version__ as kernprof_version	
except Exception:	
    kernprof_version = 'UNKNOWN_VERSION'	
parser = optparse.OptionParser(usage=usage, version="%prog " + kernprof_version)
```


OR option (b): 
```python
from line_profiler import __version__ as kernprof_version	
parser = optparse.OptionParser(usage=usage, version="%prog " + kernprof_version)
```

My vote would be for option (a), and I know you would vote for (b). If anyone else (@ta946) cares to cast a vote, I'll defer to the majority. If nobody responds I flipped a tie-breaker coin and the result was (b).

Reply:
i would vote for the try except to avoid the whole module not working just because of the version.

Regarding the interaction between the 2 modules. kernprof is more like a profiling manager, where you can choose which profiler you want to profile your code and kernprof runs it all. while line_profiler is the profiler itself.

So you can use kernprof without line_profiler (with other profilers), whereas line_profiler cannot be used without kernprof.
Because of that, i dont believe making kernprof tied to line_profiler is a good idea

Reply:
I re-added the try/except

Reply:
For other ways to store the version: I saw that some package use a VERSION.txt file (pytorch), a version.py file that can be imported once the package is installed but that is manually parsed/execed for the setup (urllib3 botocore requests colorama jinja), copy the value (pyyaml), use versioneer (pandas). I think in any case they all require to be installed/cloned entirely as the version is stored in another file.

Reply:
Yeah, I've seen that `version.py` pattern. The thing I don't like about it is that the package needs to be installed. I really like being able to just cd into a repo directory and immediately import the module. I imagine there is probably a way to rectify the two paradigms, such that if the module is installed it uses the `version.py` which might do some git magic, and if its not have some "uninstalled" hardcoded fallback. That's a separate issue though. 

For now this LGTM.

Reply:
Regarding reading from the version file on import without installing, just thought of the idea so idk if its feasible:
in the __init__.py
Can open the version file and save the value into the modules’ global space?

Reply:
I think it depends on where the version file lives. That is probably a component of the solution.